src/main/java/com/spring/board/springboard/user/service/MemberService.java

```diff
@@ -1,26 +1,119 @@
         this.memberRepository = memberRepository;
     }
 
-    public Member find(Integer memberId){
+    @Transactional(readOnly = true)
+    public Member findById(Integer memberId) {
         return memberRepository.findById(memberId)
                 .orElseThrow(() -> {
                     throw new NoMemberException(
                             MessageFormat.format("No Member data by id : {0}", memberId));
                 });
     }
+
+    @Transactional(readOnly = true)
+    public MemberResponseDto findById(Integer memberId, HttpServletRequest request) {
+
+        Cookie userInfo = getUserInfoCookie(request);
+
+        final Member findMember = memberRepository.findById(memberId)
+                .orElseThrow(() -> {
+                    throw new NoMemberException(
+                            MessageFormat.format("No Member data by id : {0}", memberId));
+                });
+
+        validateMemberByCookie(userInfo, findMember);
+
+        return new MemberResponseDto(findMember);
+    }
+
+    @Transactional
+    public void register(MemberRequestDto memberRequestDto) {
+        final Member newMember = memberRequestDto.toEntity();
+        memberRepository.save(newMember);
+    }
+
+    @Transactional
+    public MemberResponseDto login(MemberLoginDto memberLoginDto, HttpServletResponse response) {
+        Member findMember = memberRepository.findByEmail(
+                        memberLoginDto.email())
+                .orElseThrow(() -> {
+                    throw new NoMemberException(
+                            MessageFormat.format("입력한 email: [{0}] 정보로 회원을 찾을 수 없습니다.", memberLoginDto.email()));
+                });
+
+        if (isNotEqual(findMember.getPassword(),
+                memberLoginDto.password())) {
+            throw new AuthenticateException("password가 올바르지 않습니다.");
+        }
+
+        Cookie cookie = setUserInfoCookie(findMember.getEmail());
+        response.addCookie(cookie);
+
+        return new MemberResponseDto(findMember);
+    }
+
+    private boolean isNotEqual(String origin, String input) {
+        return !Objects.equals(origin, input);
+    }
+
+    private void validateMemberByCookie(Cookie cookie, Member findMember) {
+        if (isNotEqual(cookie.getValue(), findMember.getEmail())) {
+            throw new AuthenticateException("권한이 없습니다.");
+        }
+    }
+
+    private Cookie getUserInfoCookie(HttpServletRequest request) {
+        if(Objects.isNull(request.getCookies())){
+            throw new AuthenticateException(NO_COOKIE);
+        }
+
+        List<Cookie> cookies = Arrays.stream(
+                        request.getCookies())
+                .toList();
+
+        return cookies.stream()
+                .filter(cookie ->
+                        Objects.equals(cookie.getName(), userInfoCookieName))
+                .findAny()
+                .orElseThrow(() -> {
+                    throw new AuthenticateException(NO_COOKIE);
+                });
```

Comment:
혹시 리스트로 한번 변환하는 이유가 있을까요? 
```suggestion
        return Arrays.stream(request.getCookies())
                .filter(cookie ->
                        Objects.equals(cookie.getName(), userInfoCookieName))
                .findAny()
                .orElseThrow(() -> {
                    throw new AuthenticateException(NO_COOKIE);
                });
```

Reply:
나름 가독성을 주기 위함이었는데, 이제보니 굳이 안그래도 될 것 같아 보이네요!!